Trimble.Modus.Components/Controls/TMNumberInput/TMNumberInput.xaml.cs

```diff
@@ -0,0 +1,231 @@
+    /// <param name="oldValue">The old value of the property.</param>
+    /// <param name="newValue">The new value of the property.</param>
+    /// <exception cref="ArgumentException">Thrown when the new minimum value is greater than the maximum value.</exception>
+    private static void OnMinValuePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMNumberInput numberInput = bindable as TMNumberInput;
+        double min = (double)newValue;
+        double max = numberInput.MaxValue;
+        if (min > max)
+        {
+            throw new ArgumentException("Min value cannot be greater than max value");
+        }
+        else if (numberInput.Value < min)
+        {
+            numberInput.Value = min;
+        }
+        numberInput.TMInputControl.ToggleRightIconState(numberInput.Value > min);
+    }
+    /// <summary>
+    /// Called when the <see cref="MaxValue"/> property is changed.
+    /// </summary>
+    /// <param name="bindable">The object that the property was changed on.</param>
+    /// <param name="oldValue">The old value of the property.</param>
+    /// <param name="newValue">The new value of the property.</param>
+    /// <exception cref="ArgumentException">Thrown when the new minimum value is greater than the maximum value.</exception>
+    private static void OnMaxValuePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        TMNumberInput numberInput = bindable as TMNumberInput;
+        double max = (double)newValue;
+        double min = numberInput.MinValue;
+        if (min > max)
+        {
+            throw new ArgumentException("Min value cannot be greater than max value");
+        }
+        else if (numberInput.Value > max)
+        {
+            numberInput.Value = max;
+        }
+        numberInput.TMInputControl.ToggleRightIconState(numberInput.Value < max);
+    }
+
+    /// <summary>
+    /// Called when the <see cref="Value"/> property is changed.
+    /// </summary>
+    private static void OnValueChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMNumberInput numberInput)
+        {
+            numberInput.TMInputControl.Text = newValue.ToString();
+        }
+    }
+
+    /// <summary>
+    /// Method to validate that the input is a number and that it is within the min and max values
+    /// </summary>
+    private async void OnTextChanged(object sender, TextChangedEventArgs e)
+    {
+        string newText = e.NewTextValue;
+        string oldText = e.OldTextValue;
+        TMInput tMInput = sender as TMInput;
+        tMInput.TextChanged -= OnTextChanged;
+        if (string.IsNullOrEmpty(newText) || newText == "-" || newText == "." || newText == "-." || newText == "," || newText == "-,")
+        {
+            tMInput.TextChanged += OnTextChanged;
+            return;
+        }
+
+        // This is a hack since the TextChanged event is fired again when we set the text
```

Comment:
Add a FIXME comment, so that it will be searchable and you will also see a warning in the warning tab. 

        // FIXME This is a hack since the TextChanged event is fired again when we set the text

Reply:
Done